jmclient/jmclient/wallet-rpc-api.yaml

```diff
@@ -0,0 +1,618 @@
+        - mixdepth
+        - amount_sats
+        - destination
+      properties:
+        mixdepth:
+          type: integer
+          example: 0
+        amount_sats:
+          type: integer
+          example: 100000000
+        destination:
+          type: string
+          example: bcrt1qu7k4dppungsqp95nwc7ansqs9m0z95h72j9mze
+    ErrorMessage:
+        type: object
+        properties:
+          message:
+            type: string
+
+  responses:
+    # Success responses
+    DirectSend-202-Accepted:
+      description: "accepted as valid request to send transaction."
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/DirectSendResponse"
+    ListUtxos-200-OK:
+      description: "successful retrieval of utxo list"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ListUtxosResponse"
+    GetAddress-200-OK:
+      description: "successful retrieval of new address"
+      content:
+        application/json:
+           schema:
+             $ref: "#/components/schemas/GetAddressResponse"
+    ListWallets-200-OK:
+      description: "successful response to listwallets"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ListWalletsResponse"
+    Session-200-OK:
+      description: "successful heartbeat response"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/SessionResponse"
+    Create-201-OK:
+      description: "wallet created successfully"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/CreateWalletResponse"
+    Unlock-200-OK:
+      description: "wallet unlocked successfully"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/UnlockWalletResponse"
+    Display-200-OK:
+      description: "wallet display contents retrieved successfully."
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/WalletDisplayResponse"
+    Lock-200-OK:
+      description: "wallet locked successfully"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/LockWalletResponse"
+    202-Accepted:
+      description: The request has been submitted successfully for processing, but the processing has not been completed.
+    204-NoResultFound:
+      description: No result found for matching search criteria.
+    # Clientside error responses
+    400-BadRequest:
+      description: Bad request. Refer to the moreinfo section for details.  Please check that you've entered the correct parameters.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorMessage'
+    401-Unauthorized:
+      description: Unable to authorise the credentials that were supplied.
+      content:
+        application/json:
```

Comment:
So if I change it to `text/html`, should I just remove the `schema` definition below?

Reply:
I'm not sure, so don't quote me on that, but it seems to me, by quickly looking at swagger, we have at least a couple of options.

We could turn the error into:
```diff
     ErrorMessage:
-        type: object
-        properties:
-          message:
-            type: string
+        type: string
```

or we could keep `application/json` type and start returning a dict like in the spec, instead of just a string. So something like `{'message': 'error message string'}`.

Reply:
Decided on the latter. Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/996/commits/e80a948ec28522370559ec699b25229726e3f549